src/WopiHost.Core/Controllers/ContainersController.cs

```diff
@@ -109,12 +94,12 @@ public async Task<IActionResult> CreateChildContainer(
             return new BadRequestResult();
         }
 
-        IWopiResource? newFolder;
+        IWopiFolder? newFolder;
 
         // "specific mode" - The host must not modify the name to fulfill the request.
         if (!string.IsNullOrWhiteSpace(relativeTarget))
         {
-            newFolder = await storageProvider.GetWopiResourceByName(WopiResourceType.Container, id, relativeTarget, cancellationToken);
+            newFolder = await storageProvider.GetWopiResourceByName(WopiResourceType.Container, id, relativeTarget, cancellationToken) as IWopiFolder;
```

Comment:
how about we use generics for this? nice to have....for later :)

```c#
    Task<T?> GetWopiResourceByName<T>(WopiResourceType resourceType, string containerId, string name, CancellationToken cancellationToken = default) where T : class, IWopiResource;

```

Reply:
I thought you would never ask 😄 
That was my intention with introducing IWopiResource and having unit tests on the big controllers - but now that you are ready for it wait for this next commit 👍

Reply:
PS: don't even need the resourceType, did it like this:
```c#
Task<T?> GetWopiResourceByName<T>(string containerId, string name, CancellationToken cancellationToken = default)
    where T : class, IWopiResource;
```

Reply:
yeah, obviously...my bad :)